persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/DefaultQueryInterceptor.java

```diff
@@ -87,9 +93,15 @@ public void register(EventListener listener) {
     Single<ResultMessage.Rows> rows =
         interceptStatement.execute(state, options, queryStartNanoTime);
     return rows.map(
-        r ->
-            new ResultMessage.Rows(
-                new ResultSet(selectStatement.getResultMetadata(), r.result.rows)));
+        r -> {
+          if (state.getClientState() instanceof ClientStateWithBoundPort) {
+            ClientStateWithBoundPort clientState =
+                (ClientStateWithBoundPort) state.getClientState();
+            replaceNativeTransportPort(r.result, clientState.boundPort());
```

Comment:
Do we not need the same fix for C* 3.11 and 4.0?

Reply:
Not for 3.11, but it could be added for `system.peers_v2` in C* 4.0, but that's not the main aim of this PR. We could add a follow up issue.

Reply:
Added: https://github.com/stargate/stargate/issues/1119